developer_docs/how_to_create_a_plugin.asciidoc

```diff
@@ -1678,6 +1678,42 @@ The settings are strongly typed and you have to define it.
 The basic types supported by Foreman are: `:boolean`, `:integer`, `:float`, `:string`, `:text`, `:hash`, `:array`.
 The `:text` type supports markdown and usage of such setting should expect markdown syntax when using it.
 
+==== Validate setting value
+
+To validate setting value, you can use API, that tries to mimic the API of ActiveRecord.
+We can use most of the perks offered by ActiveRecord, only defining on setting name instead of attribute.
+We are adding just some shothands like direct regexp validations.
+The attribute is always `value`, you can't validate anything else as it is the only user input.
+Custom validation methods allow you to run validations against other attributes, but please dont! :)
```

Comment:
There are legitimate use cases for this. For example, for console encryption we have a field `..._key` and `..._cert` but if the whole feature is off, it's allowed to be empty. If it's on, it must be set. Right now we have no way to properly express this.

Reply:
Other attributes mean description, full_name, options etc.
I'll make it more clear! :)